lib/elixir/src/elixir_erl.erl

```diff
@@ -548,6 +549,33 @@ signature_to_binary(Module, '__struct__', []) ->
 signature_to_binary(_, Name, Signature) ->
   'Elixir.Macro':to_string({Name, [], Signature}).
 
+checker_chunk(#{definitions := Definitions, deprecated := Deprecated, compile_opts := CompileOpts}) ->
+  DeprecatedMap = maps:from_list(Deprecated),
+
+  Exports =
+    lists:foldl(fun({Function, Kind, _Meta, _Clauses}, Acc) ->
+      case Kind of
+        _ when Kind == def orelse Kind == defmacro ->
+          Reason = maps:get(Function, DeprecatedMap, nil),
+          [{Function, #{kind => Kind, deprecated_reason => Reason}} | Acc];
+        _ ->
+          Acc
+      end
+    end, [], Definitions),
+
+  Contents = #{
+    exports => lists:sort(Exports),
+    no_warn_undefined => no_warn_undefined(CompileOpts)
```

Comment:
We don't need this information in here. This information is only used when the module is compiled (when this module is the caller) and the information in this chunk is only used when this module is called. I think the contents of the chunk should be: `[{<<"ExCk">>, erlang:term_to_binary({elixir_checker_v1, Exports})}].`

Reply:
The chunk is used by runtime modules (modules that call compiled modules at runtime), so it is the other way around, the chunk is used when the module is the caller. We need to check runtime modules because the modules they call may have changed and removed functions that should cause undefined warnings. If we remove `no_warn_undefined` then we lose the ability for disabling the warnings.

BUT, it seems you are right after all, because the module adding `@compile {:no_warn_undefined, Foo}` adds a compile time dependency on `Foo`. So modules that disable the warning will never be runtime modules.

Reply:
No, I am wrong then because Bar can change on runtime for other reasons, leading it to be reverified, even if Foo did not change. It is also partially annoying that @compile adds a compile dependency but that’s rarely used within modules in the same app, so we are likely fine.